packages/fast-jss-manager-react/src/manage-jss.tsx

```diff
@@ -109,44 +81,11 @@ function manageJss<S, C>(styles?: ComponentStyles<S, C>): <T>(Component: React.C
             constructor(props: T) {
                 super(props);
 
-                // On construction, check if the style object or component object have already been used.
-                // If not, we need to store them in our weakmaps for later use
-                if (!Boolean(JSSManager.styleMap.get(styles))) {
-                    JSSManager.styleMap.set(styles, uniqueId());
-                }
-
-                if (!Boolean(JSSManager.componentMap.get(Component))) {
-                    JSSManager.componentMap.set(Component, uniqueId());
-                }
-
-                const styleId: string = JSSManager.styleMap.get(styles);
-                const componentId: string = JSSManager.componentMap.get(Component);
-
-                // Store the instance id as a combination of the generated IDs
-                this.instanceId = `${componentId}${styleId}`;
-                let separatedStylesInstance: ISeparatedStylesheet<S, C> = JSSManager.separatedStyles[this.instanceId];
-
-                // Check if we have a separatedStyles object stored at the instanceId key.
-                // If we don"t, we need to create that object
-                if (!Boolean(separatedStylesInstance)) {
-                    separatedStylesInstance = JSSManager.separatedStyles[this.instanceId] = this.separateStyles(styles);
-                }
+                JSSManager.styles = styles;
```

Comment:
Yea I don't think you need to store this anymore - you an just reference `styles` throughout.

With this implementation, you're actually clobbering a static value for every instance of a component